client/asset/btc/btc.go

```diff
@@ -821,55 +822,76 @@ func (btc *ExchangeWallet) FundingCoins(ids []dex.Bytes) (asset.Coins, error) {
 		pt := newOutPoint(txHash, vout)
 		fundingCoin, found := btc.fundingCoins[pt]
 		if found {
-			coins = append(coins, newOutput(btc.node, txHash, vout, fundingCoin.amount, fundingCoin.redeemScript))
+			coins = append(coins, newOutput(btc.node, txHash, vout, fundingCoin.amount))
 			continue
 		}
-		notFound[pt] = struct{}{}
+		notFound[pt] = true
 	}
-
 	if len(notFound) == 0 {
 		return coins, nil
 	}
-	_, utxoMap, _, err := btc.spendableUTXOs(0)
+
+	// Check locked outputs for not found coins.
+	lockedOutpoints, err := btc.wallet.ListLockUnspent()
 	if err != nil {
 		return nil, err
 	}
-
-	lockers := make([]*output, 0, len(ids))
-
-	for _, id := range ids {
-		txHash, vout, err := decodeCoinID(id)
+	for _, rpcOP := range lockedOutpoints {
+		txHash, err := chainhash.NewHashFromStr(rpcOP.TxID)
 		if err != nil {
-			return nil, err
+			return nil, fmt.Errorf("error decoding txid from rpc server %s: %v", rpcOP.TxID, err)
 		}
-		pt := newOutPoint(txHash, vout)
+		pt := newOutPoint(txHash, rpcOP.Vout)
+		if notFound[pt] {
+			txOut, err := btc.node.GetTxOut(txHash, rpcOP.Vout, true)
+			if err != nil {
+				return nil, fmt.Errorf("gettxout error for locked outpoint %v: %v", pt.String(), err)
+			}
+			var address string
+			if len(txOut.ScriptPubKey.Addresses) > 0 {
+				address = txOut.ScriptPubKey.Addresses[0]
+			}
+			utxo := &utxo{
+				txHash:  txHash,
+				vout:    rpcOP.Vout,
+				address: address,
+				amount:  toSatoshi(txOut.Value),
+			}
+			coin := newOutput(btc.node, txHash, rpcOP.Vout, toSatoshi(txOut.Value))
+			coins = append(coins, coin)
+			btc.fundingCoins[pt] = utxo
+			delete(notFound, pt)
+			if len(notFound) == 0 {
+				return coins, nil
+			}
+		}
+	}
+
+	// Some funding coins still not found after checking locked outputs.
+	// Check wallet unspent outputs as last resort. Lock the coins if
+	// they're found.
+	_, utxoMap, _, err := btc.spendableUTXOs(0)
+	if err != nil {
+		return nil, err
+	}
+	unlockedOutputs := make([]*output, 0, len(ids))
+	for pt := range notFound {
 		utxo, found := utxoMap[pt]
 		if !found {
 			return nil, fmt.Errorf("funding coin %s not found", pt.String())
 		}
-		btc.fundingCoins[pt] = utxo
-		coin := newOutput(btc.node, utxo.txHash, utxo.vout, utxo.amount, utxo.redeemScript)
+		btc.fundingCoins[pt] = utxo.utxo
+		coin := newOutput(btc.node, utxo.txHash, utxo.vout, utxo.amount)
 		coins = append(coins, coin)
-		lockers = append(lockers, coin)
+		unlockedOutputs = append(unlockedOutputs, coin)
```

Comment:
`locker` does sound like a locker collection, more confusing to me than `unlockedOutputs`. But `unlockedOutputs` does sound like a list of already unlocked outputs, this later became obvious to me as well when re-reading the code.

You're right that this-like refactors have nothing to do with the stated goals of this PR though. But I want to change that please. I worry that simply making the changes required for the stated goals causes most of the affected code to become more difficult to follow through and bugs could slip in.

Discussed with @chappjc previously about making FundOrder return `[]struct{Coin, dex.Bytes}` rather than `[]Coin, []dex.Bytes` to prevent implementations from _erroneously_ returning unequal number of coins and redeem scripts, leading to runtime index out of range panics.

Making that last change is making the related code uncomfortably more complex but I've been able to settle on a refactor that is both simplified and similar for btc and dcr, which should make the affected code generally more readable and maintainable.

Reply:
I'm not too concerned about such bugs with different numbers of coins and scripts, but I acknowledge the motivation for this approach.

Reply:
I fully support a new `struct{Coin, dex.Bytes}` **asset** type.

As far as the variable rename, I'm failing to see how it might prevent bugs, and I don't think that `lockers` was too bad to begin with. If you really feel the need to change it though, just pick something that improves readability. Maybe `coinsToLock` or similar.

Reply:
The bug prevention bit was regarding returning 2 slices from FundOrder, not the variable rename. `coinsToLock` does sound better btw.